src/v6tools/Subnet.php

```diff
@@ -62,4 +62,41 @@ public function isInSubnet($ipv6addr) {
         }
         return true;
     }
+
+    /**
+     * Generates EUI64 address
+     *
+     * @param string Mac address
+     */
+    public function getEUI64Address($mac) {
+        // Validate prefix length
+        if ($this->preflen != 64) {
+            throw new \InvalidArgumentException("Prefix length is not 64.");
+        }
+
+        // Validate mac address
+        if (!filter_var($mac, FILTER_VALIDATE_MAC)) {
+            throw new \InvalidArgumentException("Invalid mac address.");
+        }
+
+        // Expand MAC address
+        $mac = explode(':', str_replace(['.', '-', ':'], ':', $mac));
+        $mac = sprintf('%04x', (hexdec($mac[0]) << 8 | hexdec($mac[1])) ^ 0x200) .
+            sprintf('%04x', hexdec($mac[2]) << 8 | 0xff) .
+            sprintf('%04x', hexdec($mac[3]) | 0xfe00) .
+            sprintf('%02x', hexdec($mac[4])) .
+            sprintf('%02x', hexdec($mac[5]));
+
+        // Expand IP address
+        $ip = unpack('n*', inet_pton($this->addr));
```

Comment:
Can we reuse IPv6Address::expand?

Reply:
@dsp  Just thinking - would it be better to return instance of IPv6Address instead of string?

Reply:
Anyway - back to original question - no problem at all.

Reply:
Good catch. If we can return an IPv6Address, that'll be great.